core/node/prover_api/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+mod error;
+mod processor;
+
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::Context;
+use axum::{extract::Path, routing::post, Json, Router};
+use tokio::sync::watch;
+use zksync_basic_types::{commitment::L1BatchCommitmentMode, L1BatchNumber};
+use zksync_config::configs::prover_api::ProverApiConfig;
+use zksync_dal::{ConnectionPool, Core};
+use zksync_object_store::ObjectStore;
+use zksync_prover_interface::api::{ProofGenerationDataRequest, VerifyProofRequest};
+
+use crate::processor::Processor;
+
+pub async fn run_server(
+    config: ProverApiConfig,
+    blob_store: Arc<dyn ObjectStore>,
+    connection_pool: ConnectionPool<Core>,
+    commitment_mode: L1BatchCommitmentMode,
+    mut stop_receiver: watch::Receiver<bool>,
+) -> anyhow::Result<()> {
+    let bind_address = SocketAddr::from(([0, 0, 0, 0], config.http_port));
+    tracing::debug!("Starting external prover API server on {bind_address}");
+    let app = create_router(blob_store, connection_pool, commitment_mode, config).await;
+
+    let listener = tokio::net::TcpListener::bind(bind_address)
+        .await
+        .with_context(|| format!("Failed binding external prover API server to {bind_address}"))?;
+    axum::serve(listener, app)
+        .with_graceful_shutdown(async move {
+            if stop_receiver.changed().await.is_err() {
+                tracing::warn!("Stop signal sender for external prover API server was dropped without sending a signal");
+            }
+            tracing::info!("Stop signal received, external prover API server is shutting down");
+        })
+        .await
+        .context("External prover API server failed")?;
+    tracing::info!("External prover API server shut down");
+    Ok(())
+}
+
+async fn create_router(
+    blob_store: Arc<dyn ObjectStore>,
+    connection_pool: ConnectionPool<Core>,
+    commitment_mode: L1BatchCommitmentMode,
+    config: ProverApiConfig,
+) -> Router {
+    let mut processor = Processor::new(
+        blob_store.clone(),
+        connection_pool.clone(),
+        commitment_mode,
+        L1BatchNumber(config.last_available_batch),
```

Comment:
This shouldn't be a part of the config. Instead, we should do a database query here and choose the value dynamically.